olive/cache.py

```diff
@@ -32,127 +37,243 @@ class CacheSubDirs:
     @classmethod
     def from_cache_dir(cls, cache_dir: Path) -> "CacheSubDirs":
         return cls(
-            models=cache_dir / "models",
+            cache_dir=cache_dir,
             runs=cache_dir / "runs",
             evaluations=cache_dir / "evaluations",
             resources=cache_dir / "resources",
             mlflow=cache_dir / "mlflow",
         )
 
 
+class CacheConfig(ConfigBase):
+    cache_dir: Union[str, Path, List[str]] = DEFAULT_CACHE_DIR
+    clean_cache: bool = False
+    clean_evaluation_cache: bool = False
+    account_name: str = None
+    container_name: str = None
+    enable_shared_cache: bool = False
+    update_shared_cache: bool = True
+
+    @validator("account_name", pre=True)
+    def validate_account_name(cls, v, values):
+        if v:
+            return v
+        match = cls._get_shared_cache_match(values.get("cache_dir"))
+        return match.group(1) if match else None
+
+    @validator("container_name", pre=True)
+    def validate_container_name(cls, v, values):
+        if v:
+            return v
+        match = cls._get_shared_cache_match(values.get("cache_dir"))
```

Comment:
Same as above. `cache_dir` can be a list of strings also and you need to handle that.

Reply:
Same as above